art-blocks-101/on-chain.md

```diff
@@ -0,0 +1,126 @@
+- The metadata can be lost if the server goes down or an admin decides not to discontinue paying for the service.
+- The metadata can be censored by the server owner.
+
+### Hash On-Chain
+
+The hash of the metadata is stored on-chain, and the metadata are stored on a server. Storing the hash of the metadata makes the NFT immutable, while keeping it much cheaper than storing all of the metadata on-chain. However, it has a few drawbacks:
+
+- The metadata can be lost if the server goes down or an admin decides not to discontinue paying for the service.
+- The metadata can be censored by the server owner.
+
+Projects such as CryptoPunks originally stored the hash of the metadata on-chain, but the metadata were stored on a server. They have since upgraded to a fully on-chain solution.
+
+### Decentralized Storage Network
+
+The metadata are stored on a decentralized storage network, such as IPFS or Arweave. This is a popular option, and enables the NFT to be immutable. Drawbacks include:
+
+- The metadata must remain pinned by at least one party in order to be accessible.
+
+### Fully On-Chain
+
+The metadata are stored on the blockchain. This is the most expensive option, but it has many advantages. The metadata are always accessible and available, as long as the NFT exists. Collectors and artists never have to worry about a token's outputs disappearing.
+
+### Software Registries
+
+Some NFTs (like some Art Blocks project tokens) may rely on published software libraries when generating outputs. In these cases, the NFT metadata may include information about dependencies and where to find them (e.g. "p5js, v1.0.0 on the npm software registry"). Software registries are a special form of decentralized storage, because of the distributed nature of the software repositories and many downloads on developers' computers worldwide.
+
+## Art Blocks Storage Solutions
+
+### Art Blocks Fully On-Chain
+
+Art Blocks Flagship and Art Blocks Engine NFTs only rely on fully on-chain storage solutions, plus allowing a single dependency to be housed on a software registry. This provides an extremely high level of confidence that the NFTs will remain fully accessible and immutable for the foreseeable future.
+
+While Art Blocks originally relied on the npm software registry to house dependencies, the Art Blocks team has since created a **fully on-chain software registry called the [Art Blocks Dependency Registry](https://github.com/ArtBlocks/artblocks-contracts/blob/main/packages/contracts/README.md#dependencyregistry)**. The registry is fully decentralized and is used by all Art Blocks Flagship NFTs. The registry enables dependencies to be stored fully on-chain, when the gas costs are justified. Art Blocks uploaded their first dependency, p5js v1.0.0, to the registry in January 2024, retiring any doubt that projects relying on p5js@1.0.0 would ever be unable to generate their outputs in the future.
```

Comment:
Super minor nit: we were pulling most of the deps from cdnjs so would maybe just say widely used CDN instead of npm specifically

Reply:
good call - updated in [21be065](https://github.com/ArtBlocks/artblocks-docs/pull/301/commits/21be0657023f9631878b11ba6637adce8de1c486)